submissions/UA-girl/task_js_oop/app.js

```diff
@@ -0,0 +1,179 @@
+    low: [200, 400],
+    medium: [300, 500],
+    hight: [200, 600]
+};
+
+let holder;
+
+const button = document.querySelector('.button');
+const settings = document.querySelector('.settings');
+
+createScoreHolder();
+
+['click', 'keyup'].forEach(item => button.addEventListener(item, submitButton));
+
+function submitButton(event) {
+    if (event.type === 'keyup' && event.keyCode !== 13) {
+        return;
+    }
+    const avatar = document.querySelector('.radio__avatar:checked');
+    const level = document.querySelector('.radio__level:checked');
+    settings.classList.toggle('hidden');
+    for (let i = 0; i < allEnemies.length; i++) {
+        allEnemies[i].setSpeed(level.value);
+    }
+    player.sprite = 'images/' + avatar.value;
+}
+
+function createScoreHolder() {
+    holder = document.createElement('div');
+    const body = document.querySelector('body');
+    holder.innerHTML = `Your current score is 0`;
+    body.appendChild(holder);
+}
+
+function getRandomNumber(minimum, maximum) {
+    return Math.floor(Math.random() * (maximum - minimum)) + minimum;
+}
+
+function checkMatchPosition(obj1, obj2) {
+    return (Math.round(obj1.x / stepX) === Math.round(obj2.x / stepX)) &&
+        (Math.round(obj1.y / stepY) === Math.round(obj2.y / stepY));
+}
+
+let Enemy = function () {
+    this.setDefaultPosition();
+    this.setSpeed();
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.setSpeed = function (key = 'low') {
+    const [min, max] = levels[key];
+    this.speed = getRandomNumber(min, max);
+};
+
+Enemy.prototype.setDefaultPosition = function () {
+    this.x = 0;
+    this.y = startY[getRandomNumber(0, 3)];
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += dt * this.speed;
+    if (this.x > areaWidth) {
+        this.setDefaultPosition();
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.checkCollisions = function (player) {
+    if (checkMatchPosition(this, player)) {
+        player.updateScore(-5);
+        holder.innerHTML = `You lose! =(
+         Your score: ${player.score}`;
+        player.setDefaultPosition();
+        allTreasures.forEach(treasure => treasure.setDefaultPosition());
+    }
+};
+
+
+let Player = function () {
+    this.setDefaultPosition();
+    this.sprite = 'images/char-boy.png';
+    this.score = 0;
+};
+
+Player.prototype.setDefaultPosition = function () {
+    [this.x, this.y] = startPlayerXY;
+};
+
+Player.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.update = function () {
+    if (this.isWin()) {
+        this.setDefaultPosition();
+        this.updateScore(5);
+        holder.innerHTML = `Cool! You won!!!
+        Your score is ${this.score}`;
+    }
+    allEnemies.forEach(enemy => enemy.checkCollisions(player));
+    allTreasures.forEach(treasure => treasure.checkCollisions(player));
```

Comment:
Here player is onbiously a variable from outside the scope of this class.
In fact you do not need and in current scope you have a simple way to refer a player not resorting to any global variables.

`allEnemies` and `allTreasures` are also global variables. Please make the `Player` class receive them, not know them.